app/controllers/import_controller.rb

```diff
@@ -26,5 +26,35 @@ def import
       end
     end
   end
-end
-  
+
+  def import_portal_activity
```

Comment:
I don't see how the user_id would work since I assume the request is coming from the Portal directly. So there would be no cookie sent from the users browser. But perhaps this doesn't work the way I think?

If  the user_id approach is secure then that is fine. But if it is not secure, then yes change the sso:add_client rake task so the domain is asked for. The domain can be optional.